pandas/tests/arrays/test_datetimelike.py

```diff
@@ -56,7 +56,68 @@ def timedelta_index(request):
     return pd.TimedeltaIndex(['1 Day', '3 Hours', 'NaT'])
 
 
-class TestDatetimeArray(object):
+class SharedTests(object):
+    index_cls = None
+
+    def test_take(self):
+        data = np.arange(100, dtype='i8')
+        np.random.shuffle(data)
+
+        idx = self.index_cls._simple_new(data, freq='D')
+        arr = self.array_cls(idx)
+
+        takers = [1, 4, 94]
+        result = arr.take(takers)
+        expected = idx.take(takers)
+
+        tm.assert_index_equal(self.index_cls(result), expected)
+
+        takers = np.array([1, 4, 94])
+        result = arr.take(takers)
+        expected = idx.take(takers)
+
+        tm.assert_index_equal(self.index_cls(result), expected)
+
+    def test_take_fill(self):
+        data = np.arange(10, dtype='i8')
+
+        idx = self.index_cls._simple_new(data, freq='D')
+        arr = self.array_cls(idx)
+
+        result = arr.take([-1, 1], allow_fill=True, fill_value=None)
+        assert result[0] is pd.NaT
+
+        result = arr.take([-1, 1], allow_fill=True, fill_value=np.nan)
+        assert result[0] is pd.NaT
+
+        result = arr.take([-1, 1], allow_fill=True, fill_value=pd.NaT)
+        assert result[0] is pd.NaT
+
+        with pytest.raises(ValueError):
+            arr.take([0, 1], allow_fill=True, fill_value=2)
+
+        with pytest.raises(ValueError):
+            arr.take([0, 1], allow_fill=True, fill_value=2.0)
+
+        with pytest.raises(ValueError):
+            arr.take([0, 1], allow_fill=True,
+                     fill_value=pd.Timestamp.now().time)
+
+    def test_concat_same_type(self):
```

Comment:
As I think also commented on one of the previous PRs that started doing this, I don't think we should test `_concat_same_type` here directly. It is already tested by the base extension tests and by all the tests that actually use it under the hood.

Reply:
Yah, most of these tests were salvaged from one of those older PRs.  I don't see much downside to having the tests, but am pretty happy to pawn this decision/PR off on Tom

Reply:
We still want the fail causes (different dtypes) here.

At this point, we should be able to simplify `core/dtypes/concat.py::concat_datetimetz` and `DatetimeIndex._concat_same_dtype` right? I'll take a look.